src/lib/utils/hooks/SynapseAPI/useEntity.ts

```diff
@@ -134,19 +138,20 @@ export function useGetJson(
 }
 
 export function useUpdateViaJson(
-  entityId: string,
-  json: unknown,
-  options?: UseMutationOptions<EntityJson, SynapseClientError>,
+  options?: UseMutationOptions<EntityJson, SynapseClientError, EntityJson>,
 ) {
   const queryClient = useQueryClient()
   const { accessToken } = useSynapseContext()
-
-  return useMutation<EntityJson, SynapseClientError>(
-    [accessToken, 'entity', entityId, 'json'],
-    () => SynapseClient.updateEntityJson(entityId, json, accessToken),
+  return useMutation<EntityJson, SynapseClientError, EntityJson>(
+    (json: EntityJson) => {
+      const entityId = json.id
+      return SynapseClient.updateEntityJson(entityId, json, accessToken)
+    },
     {
       ...options,
       onSuccess: async (data, variables, ctx) => {
+        const entityId = data.id
+
         await queryClient.invalidateQueries([accessToken, 'entity', entityId], {
           exact: false,
         })
```

Comment:
Removing the mutation key because it doesn't seem to be very useful (and it's causing me to get new typescript errors which are breaking the build). Also, now we should call `mutation.mutate` with new data, instead of passing the new data into the hook fn.